[guides] Add a guide on writing document titles and headings (#32199)
expo/expo: guides/Wrtiting scannable titles and headings.md

# Writing document titles and headings optimized for search

In Expo documentation, document _titles_ and section _headings_ provide structure and reference points to help developers search and scan any document. Part of our job is to help them reach the point they need to solve their problem.

Following suggestions and tips from this guide will help you to write document titles and section headings that are easy to search and make a document scannable.

---

Let's start with an example where a heading is unaligned to its document's content:

[<img src="./assets/writing-guides/01-search-doc-example.png" width="540" />](./assets/writing-guides/01-search-doc-example.png)

Based on the search term above, the correct page appears as the last item. The search term contains all keywords that are common with the way we talk about running an EAS Build on a local computer.

> Using the context of a keyword (commonly used term), it improves the search ranking of the document when the same search term is used:

Reviewer: ```suggestion
Using keywords in the heading improves the search ranking of the document when the same search term is used:
```

Author: not exactly; I meant to say here that some sort of context that represents the keyword. A keyword in every case will not be an ideal fit. I'll try rephrase it.

[<img src="./assets/writing-guides/02-improved-search-doc-example.png" width="540" />](./assets/writing-guides/02-improved-search-doc-example.png)

The improvement above also tells the developer about the primary purpose of the document on first glance.

## General tips

When writing a document title or a section heading, do the following:

- Be concise.
- Describe the main purpose of the document or the section

> - For procedural documents (guides and tutorials), use a verb-based title/heading to describe the action that you expect the developer to perform. Using an action verb also informs the developer about the type of content they can expect to find once they land on the page and start scanning it.

Reviewer: ```suggestion
- For procedural documents (guides and tutorials), use a verb-based title/heading (e.g., "Run EAS Build") to describe the action that you expect the developer to perform. Using an action verb also informs the developer about the type of content they can expect to find once they land on the page and start scanning it.
```

- Front-load keywords in headings to keep the important information or subject at the beginning.
- Focus on what matters most to customers and choose words they would use themselves.

> - Use sentence case for a section or sub-section headings. For more details, see [formatting guidelines for headings](./Expo%20Documentation%20Writing%20Style%20Guide.md#headings)

Reviewer: ```suggestion
- Use sentence case (i.e., capitalize the first word and proper nouns only) for section or sub-section headings. For more details, see [formatting guidelines for headings](./Expo%20Documentation%20Writing%20Style%20Guide.md#headings)
```

Author: I think it's more than just proper nouns; it's also branding and product/service names. This is why I linked to writing style guide doc's section: https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md#headings

### Document titles

> When using multiple content types in the same document, such as a section or two to describe a procedure that is followed by a section that explains the concept, use a task-based title (action-based verbs) for that document.

Reviewer: If task-based title is the same as the "verb-based" title/heading above, let's us the same words here.

> When using multiple content types in the same document, such as a section or two to describe a procedure that is followed by a section that explains the concept, use a task-based title (action-based verbs) for that document.

Author: ```suggestion
When using multiple content types in the same document, such as a section or two describing a procedure followed by a section explaining the concept, use a verb-based title for that document.
```
